auth-service/src/main/kotlin/uk/gov/communities/delta/auth/utils/RegistrationService.kt

```diff
@@ -0,0 +1,155 @@
+package uk.gov.communities.delta.auth.utils
+
+import OrganisationService
+import org.slf4j.LoggerFactory
+import uk.gov.communities.delta.auth.config.DeltaConfig
+
+class RegistrationService(
+    private val deltaConfig: DeltaConfig,
+    private val organisationService: OrganisationService,
+    private val emailService: EmailService
+) {
+    class RegistrationResult {
+        enum class Status {
+            CREATED, ALREADY_EXISTS, OTHER_FAILURE
+        }
+
+        val status: Status
+        val email: String
+        val token: String?
+
+        constructor(
+            status: Status,
+            email: String
+        ) {
+            this.status = status
+            this.email = email
+            token = null
+        }
+
+        constructor(
+            status: Status,
+            email: String,
+            token: String
+        ) {
+            this.status = status
+            this.email = email
+            this.token = token // TODO - are these still needed with tokens stored in db (see below)
+        }
+    }
+
+    // Previously users could be datamart users but not delta users, at migration we only transferred over users who
+    // were delta users so there should never be users who exist but aren't delta users - therefore can remove all logic
+    // for attaching users and checking if they are indeed delta users (if they somehow existed they'd get a "contact
+    // the service desk" error at login which is what we want
+
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+    private val userService = UserService()
+
+    //        val context = Context(Locale.getDefault(), model) TODO - what did this do? Is it still needed? - from email sending
+
+
+    fun register(registration: Registration): RegistrationResult { // TODO - does this still need to return anything?
+        val user = userService.findUserById(registration.emailAddress, true).orElse(null)
+        if (user != null) {
+            // TODO - logic check - do we want this to happen?
+            //  - if a user exists already then registering them again will add them to organisations associated with their domain
+            //      - can't see a scenario where this isn't already the case but if it is then do we definitely want to fix it?
```

Comment:
I don't think so, should just say the account exists and send them to the login page. It kinda made sense when there were non-delta users in datamart, but not any more.

Reply:
Yay will delete